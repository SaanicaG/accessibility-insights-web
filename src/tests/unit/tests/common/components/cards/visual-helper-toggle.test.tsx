// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { userEvent } from '@testing-library/user-event';

import { render } from '@testing-library/react';
import { VisualHelperToggle } from 'common/components/cards/visual-helper-toggle';
import { AutomatedChecksCardSelectionMessageCreator } from 'common/message-creators/automated-checks-card-selection-message-creator';
import * as React from 'react';
import { It, IMock, Mock } from 'typemoq'; 

describe('VisualHelperToggle', () => {
    let mockCardSelectionMessageCreator: IMock<AutomatedChecksCardSelectionMessageCreator>;

    beforeEach(() => {
        mockCardSelectionMessageCreator = Mock.ofType(AutomatedChecksCardSelectionMessageCreator);
    });

    it.each([true, false])(
        'renders per snapshot with visualHelperEnabled %s',
        (visualHelperEnabled: boolean) => {
            const renderResult = render(<VisualHelperToggle
                visualHelperEnabled={visualHelperEnabled}
                cardSelectionMessageCreator={mockCardSelectionMessageCreator.object}
            />);
            expect(renderResult.asFragment()).toMatchSnapshot();
        },
    );

    it('fires toggleVisualHelper when toggled', async () => {

              mockCardSelectionMessageCreator                       
            .setup(m => m.toggleVisualHelper(It.isAny())) ;    
                                                              

        const renderResult = render(<VisualHelperToggle
            visualHelperEnabled={false}
            cardSelectionMessageCreator={mockCardSelectionMessageCreator.object}
        />);
        await userEvent.click(renderResult.getByRole('switch')); 

        mockCardSelectionMessageCreator.verifyAll();
    });
});
