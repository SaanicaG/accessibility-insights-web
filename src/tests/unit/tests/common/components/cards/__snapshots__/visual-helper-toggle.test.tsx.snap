// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VisualHelperToggle renders per snapshot with visualHelperEnabled false 1`] = `
<DocumentFragment>
  <div
    class="ms-Toggle is-enabled visualHelperToggle root-109"
  >
    <label
      class="ms-Label ms-Toggle-label label-115"
      for="Toggle1"
      id="Toggle1-label"
    >
      Visual helper
    </label>
    <div
      class="ms-Toggle-innerContainer container-111"
    >
      <button
        aria-checked="false"
        aria-labelledby="Toggle1-label"
        class="ms-Toggle-background pill-116"
        data-is-focusable="true"
        data-ktp-target="true"
        id="Toggle1"
        role="switch"
        type="button"
      >
        <span
          class="ms-Toggle-thumb thumb-117"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`VisualHelperToggle renders per snapshot with visualHelperEnabled true 1`] = `
<DocumentFragment>
  <div
    class="ms-Toggle is-checked is-enabled visualHelperToggle root-109"
  >
    <label
      class="ms-Label ms-Toggle-label label-115"
      for="Toggle0"
      id="Toggle0-label"
    >
      Visual helper
    </label>
    <div
      class="ms-Toggle-innerContainer container-111"
    >
      <button
        aria-checked="true"
        aria-labelledby="Toggle0-label"
        class="ms-Toggle-background pill-112"
        data-is-focusable="true"
        data-ktp-target="true"
        id="Toggle0"
        role="switch"
        type="button"
      >
        <span
          class="ms-Toggle-thumb thumb-113"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
